ddtrace/contrib/botocore/patch.py

```diff
@@ -376,9 +381,48 @@ def patched_api_call(original_func, instance, args, kwargs):
                     if endpoint_name == "lambda" and operation == "Invoke":
                         inject_trace_to_client_context(params, span)
                     if endpoint_name == "sqs" and operation == "SendMessage":
-                        inject_trace_to_sqs_or_sns_message(params, span, endpoint_name)
+                        data_streams_context = None
```

Comment:
I'd do three things here:

1. Put the core logic into another function
2. Mirror the `inject_trace_to_sqs_or_sns` model and have the batch method call ^ for each entry
3. Rename the `inject_trace_data_to_message_attributes` method to something like `inject_datadog_data_to_message_attributes` and use it for both DSM (what you wrote) and APM (what already exists)

This will get rid of a lot of the repeat code and follow the already-existing pattern.